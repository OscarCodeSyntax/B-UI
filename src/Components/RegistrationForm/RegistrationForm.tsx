import FormControl from "@mui/material/FormControl";
import React, { useState } from "react";
import "../../App.css";
import { FormLabel, Link, TextField } from "@mui/material";
import { SignUpUserType } from "../../Resources/Types/UserLoginTypes";
import UserLoginQueries from "../../Resources/Queries/UserLoginQueries";

const RegistrationForm = () => {
  const [username, setUsername] = useState("");

  const [password, setPassword] = useState("");

  const [email, setEmail] = useState("");

  function validateForm() {
    return username.length > 0 && password.length > 0 && email.length;
  }

  const handleSubmit = async (event: any) => {
    event.preventDefault();

    let loginInformation: SignUpUserType = {
      username: username,
      email: email,
      roles: [],
      password: password,
    };
    // send the username and password to the API
    const response = await UserLoginQueries.signUpUser(loginInformation);
    // store the user in localStorage
    localStorage.setItem("username", response.username);
    localStorage.setItem("id", response.id);
    console.log(response.id);
    console.log(response.username);
    console.log(response.headers);
  };

  return (
    <form onSubmit={handleSubmit}>
      <h2>Registration</h2>
      <FormControl>
        <FormLabel>Username</FormLabel>
        <TextField
          type="username"
          color="secondary"
          onChange={(e) => setUsername(e.target.value)}
          name="username"
          value={username}
        />
        <FormLabel>Email</FormLabel>
        <TextField
          type="email"
          color="secondary"
          onChange={(e) => setEmail(e.target.value)}
          name="email"
          value={email}
        />
        <FormLabel>Password</FormLabel>
        <TextField
          type="password"
          color="secondary"
          onChange={(e) => setPassword(e.target.value)}
          name="password"
          value={password}
        />
        <br></br>
        <TextField
          type={"submit"}
          label={"Submit button"}
          color="secondary"
          value="submit"
          name="submit"
          disabled={!validateForm()}
        />
        <Link href="/" variant="body2" className="padding-10 text-center">
          Login page
        </Link>
      </FormControl>
    </form>
  );
};

export default RegistrationForm;
