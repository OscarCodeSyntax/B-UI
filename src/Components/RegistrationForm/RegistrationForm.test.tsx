import React from "react";

import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import RegistrationForm from "./RegistrationForm";

describe("LoginForm  ", () => {
  it("renders correctly", () => {
    const ApiButtonsWrapper = renderer.create(<RegistrationForm />).toJSON();
    expect(ApiButtonsWrapper).toMatchSnapshot();
  });

  it("renders all expected elements", () => {
    const testRenderer = renderer.create(<RegistrationForm />);
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("input").length).toEqual(4);
    expect(testInstance.findAllByType("label").length).toEqual(4);
    expect(testInstance.findAllByType("h1").length).toEqual(1);
  });

  it("has all the correct text", () => {
    render(<RegistrationForm />);
    expect(screen.getByText("Registration")).toBeTruthy();
    expect(screen.getByText("Username")).toBeTruthy();
    expect(screen.getByText("Password")).toBeTruthy();
    expect(screen.getByText("Email")).toBeTruthy();
    expect(screen.getByLabelText("Submit button")).toBeTruthy();
  });
});
