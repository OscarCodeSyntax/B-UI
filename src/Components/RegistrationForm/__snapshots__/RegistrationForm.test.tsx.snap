// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm   renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Registration
  </h2>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
    >
      Username
    </label>
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="username"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
    >
      Email
    </label>
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r1:"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="email"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
    >
      Password
    </label>
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r2:"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-179z3gj-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        htmlFor=":r3:"
        id=":r3:-label"
      >
        Submit button
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary Mui-disabled MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={true}
          id=":r3:"
          name="submit"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="submit"
          value="submit"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-14lo706"
          >
            <span>
              Submit button
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <a
      className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways padding-10 text-center css-101ca9i-MuiTypography-root-MuiLink-root"
      href="/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      Login page
    </a>
  </div>
</form>
`;
